scwidgets/converters.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Matthew Andrew
+
+import scipp as sc
+import pathlib
+
+scipp_object_mutable = (sc.DataArray, sc.Dataset, sc.Variable,
+                        sc.DataArrayView, sc.DatasetView, sc.VariableView)
+
+scipp_object = scipp_object_mutable + (
+    sc.DataArrayConstView, sc.DatasetConstView, sc.VariableConstView)
+
+
+def filepath_converter(filepath, data_directory=pathlib.Path.cwd()):
+    """
+    Checks the validity of the filepath provided,
+    returns it is valid. Otherwise tries to form a valid
+    filepath assuming a filename has been provided instead.
+
+    Throws if file not found.
+    """
+    path = pathlib.Path(filepath)
+
+    if path.is_file():
+        return filepath
+
+    path = data_directory / path
+
+    if path.is_file():
+        return str(path)
+
+    raise ValueError(f'Filepath {filepath} was not found.')
+
+
+def scope_converter(object_name, scope):
+    """
+    Attempts to convert the string input into an object by
+    assuming it names an object in the provided scope.
+
+    Throws if not object of correct name found.
+    """
+    if object_name in scope:
+        return scope[object_name]
+
+    raise ValueError(
+        f'Object of name {object_name} does not exist within scope.')
+
+
+def typed_scope_converter(object_name, allowed_types, scope):
```

Comment:
I understand the need for supporting only scipp objects, but what about the scope? Why is that needed.

I suggest to replace `typed_` by `scipp_` or something like that?

Reply:
Being able to specify that you expect an input to be a certain type feels like something which could have more generic usage than just scipp? 

The scope is needed because internally this converter converts a string into the object it names in the given scope.

Reply:
Right, I somehow misread this as only working for scipp objects.